### PR TITLE
[ENH] :sparkles:  Password configuration can be customized at a userstorage level

### PR DESCRIPTION
**Pull request by mehdimaaref7**
…torage level

Signed-off-by: Mehdi Maaref <mehdi.maaref@insee.fr>

### CODE REVIEW

File: docs/realm-configuration.md [hunk 1]
```diff
@@ -91,10 +91,9 @@ A list of custom key/values can be added at the end.
...
 | applicationsMaxOutputSize           | 100 |                                                                                     yes |  fr.insee.sugoi.applications.maxoutputsize | The maximum number of applications outputs allowed                                                                                                                                                                         |  
 | organizationsMaxOutputSize          | 100 |                                                                                     yes | fr.insee.sugoi.organizations.maxoutputsize | The maximum number of organizations outputs allowed                                                                                                                                                                        |  
+#### Userstorage configuration properties on password
```
Comment (CChemin) [inline docs/realm-configuration.md:94]: move it down in ## UserStorage configuration and don't forget to update the table of content at the top of this file

File: sugoi-api-model/src/main/java/fr/insee/sugoi/model/Realm.java [hunk 2]
```diff
@@ -123,7 +123,11 @@ public Map<String, String> getProperties() {
   }
 
   public Optional<UserStorage> getUserStorageByName(String name) {
-    return userStorages.stream().filter(s -> s.getName().equalsIgnoreCase(name)).findFirst();
+    if (userStorages != null) {
+      return userStorages.stream().filter(s -> s.getName().equalsIgnoreCase(name)).findFirst();
+    } else {
+      return Optional.empty();
+    }
```
Comment (CChemin) [inline sugoi-api-model/src/main/java/fr/insee/sugoi/model/Realm.java:130]: can you move it in its own commit ?

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/CredentialsServiceImpl.java [hunk 3]
```diff
@@ -56,28 +57,39 @@ public ProviderResponse reinitPassword(
     try {
       User user = userService.findById(realm, userStorage, userId);
 
-      Map<String, String> realmProperties = configService.getRealm(realm).getProperties();
```
Comment (CChemin) [inline sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/CredentialsServiceImpl.java:59]: this should use RealmProvider instead of ConfigService

File: docs/realm-configuration.md [hunk 4]
```diff
@@ -91,10 +91,9 @@ A list of custom key/values can be added at the end.
...
 
-#### Realm configuration properties on password
```
[the remark this answers is not in the document]
Comment (clement-dufaure) [reply docs/realm-configuration.md:95]: Explain explicitly that theses props can be set 
- by "global" default in props like fr.insee.sugoi.password.create.withUpperCase
override by - a value at realm level
override by - a value at userstorage level

**Review by clement-dufaure — APPROVED** [review]
